### ISSUE TITLE
Add beforeSendTransaction option

### ISSUE THREAD

**Issue by philipphofmann** [getsentry/sentry-cocoa#3339]
### Description

Add a `beforeSendTransactions` callback to the Options, as we have it for [Java](https://github.com/getsentry/sentry-java/blob/c9ca2e26898f7a44705e1a5033e48d657cb59ab1/sentry/src/main/java/io/sentry/SentryOptions.java#L709-L717). Although, transactions go through beforeSend, we plan on reverting this behavior in the next major version with https://github.com/getsentry/sentry-cocoa/issues/1730.

**Comment by sentryadam0000345** [comment]
@philipphofmann Hi Philip, do we have any updates here? Thanks

**Comment by philipphofmann** [comment]
No, sorry, not yet. You can filter transactions in `beforeSend` until we add this feature, @sentryadam0000345.

**Comment by kahest** [comment]
Adding this means not sending transactions to `beforeSend` anymore, which makes this a breaking change

**Comment by philprime** [comment]
Completed with #8745
